numpy/core/src/multiarray/ctors.c

```diff
@@ -950,6 +949,8 @@ PyArray_NewFromDescr_int(PyTypeObject *subtype, PyArray_Descr *descr, int nd,
         }
     }
 
+    /* Check dimensions */
+    size = sd;
```

Comment:
I'm OK with the name, I'm just amazed at how simply you have solved this: I took a look at this a couple of days ago and didn't find any clean way of fixing it.